### PR TITLE
Add missing isAnchorSpawn to PlayerPostRespawnEvent

### PR DESCRIPTION
**Pull request by HexedHero**
Follows the same logic as the normal PlayerRespawnEvent and works inline.
Tested with

    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
    public void a(PlayerRespawnEvent event) {
        Bukkit.getLogger().warning("Fired: " + event.isBedSpawn() + " " + event.isAnchorSpawn());
    }

    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
    public void b(PlayerPostRespawnEvent event) {
        Bukkit.getLogger().warning("POST Fired: " + event.isBedSpawn() + " " + event.isAnchorSpawn());
    }

### CODE REVIEW

**Review by DenWav — DISMISSED** [review]

File: Spigot-Server-Patches/0339-Implement-PlayerPostRespawnEvent.patch [hunk 1]
```diff
@@ -39,7 +39,7 @@ index ea6c5c9305994e64831c5444666c22a96f9d36da..ae0aaaf129c44901ac72c67e50c66333
 +
 +        // Paper start
 +        if (isRespawn) {
-+            cserver.getPluginManager().callEvent(new com.destroystokyo.paper.event.player.PlayerPostRespawnEvent(entityplayer.getBukkitEntity(), location, isBedSpawn));
++            cserver.getPluginManager().callEvent(new com.destroystokyo.paper.event.player.PlayerPostRespawnEvent(entityplayer.getBukkitEntity(), location, isBedSpawn && !flag2, flag2));
```
Comment (kennytv) [inline Spigot-Server-Patches/0339-Implement-PlayerPostRespawnEvent.patch]: This is not correct. `flag2` is false for end credit respawns, but those still teleport you to the anchor as usual (if `flag3` is set). flag2 is only used for depletion of the anchor.

**Review by kennytv — CHANGES_REQUESTED** [review]
Please also add a `isAnchorSpawn` boolean next to `isBedSpawn` so it's easier to update

File: Spigot-Server-Patches/0339-Implement-PlayerPostRespawnEvent.patch (hunk 1 above)
Comment (HexedHero) [reply Spigot-Server-Patches/0339-Implement-PlayerPostRespawnEvent.patch]: I just copied how its handled in the Bukkit's respawn event "PlayerRespawnEvent respawnEvent = new PlayerRespawnEvent(respawnPlayer, location, isBedSpawn && !flag2, flag2);"

Re-tested and you are indeed correct, seems the normal PlayerRespawnEvent event is also bugged in this way.

[12:47:12 WARN]: Fired: true false
[12:47:12 WARN]: POST Fired: true false
When entering the End portal with or without viewing the credits. Both think it's a bed spawn.

Worth noting I didn't lose a charge on my anchor despite respawning there after viewing the credits, killed my self normally and both events came out as `false true` and lost a charge as normal.

Comment (kennytv) [reply Spigot-Server-Patches/0339-Implement-PlayerPostRespawnEvent.patch]: Yeah not losing the charge is intended after the end credits as far as I can tell. As for the Bukkit issue, you could either open an issue/a pr on Spigot or another pr for the fix on Paper, however you'd like :p

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
Needs to be rebased

**Comment by HexedHero** [comment]
Rebased - First time I've had to move a commit but I think I made sure everything is in their correct patch.

**Comment by HexedHero** [comment]
Should add RespawnFlags just like how PlayerRespawnEvent works now instead of this.

**Pull request closed without merging**